src/momento/incubating/simple_cache_client.py

```diff
@@ -36,32 +39,57 @@ def dictionary_set(
         self,
         cache_name: str,
         dictionary_name: str,
-        dictionary: Dictionary,
-    ) -> CacheDictionarySetResponse:
+        key: Optional[DictionaryKey] = None,
+        value: Optional[DictionaryValue] = None,
+        dictionary: Optional[Dictionary] = None,
```

Comment:
> Is this only to reduce the quantity of sdk methods by 1? The service certainly will not support the other 6 ways.

> If you have 2 legal ways to invoke the api, why not provide 2 named ways to call that api? dictionary_set and dictionary_multi_set seem appropriate with non-optional arguments, no?

Indeed there are really two functions. I wanted to express the idea of "setting a dictionary" with the same name, like you would in a strongly typed language (eg Java). In Java you would accomplish this through function overloading. Since Python doesn't support overloading, you either use keyword arguments (like I did), or create separate functions with distinct names.

> To the maximum degree that makes sense, our SDKs should be guessable!